File: src/main/java/org/activecraft/activecraftcore/commands/RepairCommand.java
package org.activecraft.activecraftcore.commands;

import org.activecraft.activecraftcore.ActiveCraftPlugin;
import org.activecraft.activecraftcore.exceptions.ActiveCraftException;
import org.activecraft.activecraftcore.exceptions.NotHoldingItemException;
import org.activecraft.activecraftcore.ActiveCraftPlugin;
import org.activecraft.activecraftcore.exceptions.ActiveCraftException;
import org.activecraft.activecraftcore.exceptions.NotHoldingItemException;
import org.bukkit.Sound;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.Damageable;
import org.bukkit.inventory.meta.ItemMeta;

import java.util.List;

public class RepairCommand extends ActiveCraftCommand {

    public RepairCommand(ActiveCraftPlugin plugin) {
        super("repair",  plugin);
    }

    @Override
    public void runCommand(CommandSender sender, Command command, String label, String[] args) throws ActiveCraftException {
        Player player = getPlayer(sender);
        checkPermission(sender);
        checkHoldingItem(player, NotHoldingItemException.ExpectedItem.ANY);
        ItemStack item = player.getInventory().getItemInMainHand();
        ItemMeta meta = item.getItemMeta();
        if (!(meta instanceof Damageable)) {
            sendMessage(sender, this.cmdMsg("cannot-be-repaired"));
            return;
        }
        ((Damageable) meta).setDamage(0);
        item.setItemMeta(meta);
        messageFormatter.addReplacement("item-displayname", item.getI18NDisplayName());
        sendMessage(sender, this.cmdMsg("repair"));
        player.playSound(player.getLocation(), Sound.BLOCK_ANVIL_USE, 1f, 1f);
    }

    @Override
    public List<String> onTab(CommandSender sender, Command command, String label, String[] args) {
        return null;
    }
}

File: src/main/java/org/activecraft/activecraftcore/commands/RepairCommand.kt
package org.activecraft.activecraftcore.commands

import org.activecraft.activecraftcore.ActiveCraftPlugin
import org.activecraft.activecraftcore.exceptions.ActiveCraftException
import org.activecraft.activecraftcore.exceptions.NotHoldingItemException
import org.bukkit.Sound
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.inventory.meta.Damageable

class RepairCommand(plugin: ActiveCraftPlugin?) : ActiveCraftCommand("repair", plugin!!) {
    @Throws(ActiveCraftException::class)
    public override fun runCommand(sender: CommandSender, command: Command, label: String, args: Array<String>) {
        val player = getPlayer(sender)
        assertCommandPermission(sender)
        assertHoldingItem(player, NotHoldingItemException.ExpectedItem.ANY)
        val item = player.inventory.itemInMainHand
        val meta = item.itemMeta
        if (meta !is Damageable) {
            sendMessage(sender, this.cmdMsg("cannot-be-repaired"))
            return
        }
        meta.damage = 0
        item.itemMeta = meta
        item.i18NDisplayName?.let { messageFormatter.addFormatterPattern("item-displayname", it) }
        sendMessage(sender, this.cmdMsg("repair"))
        player.playSound(player.location, Sound.BLOCK_ANVIL_USE, 1f, 1f)
    }

    public override fun onTab(
        sender: CommandSender,
        command: Command,
        label: String,
        args: Array<String>
    ) = null
}